node/src/container_chain_spawner.rs

```diff
@@ -238,6 +240,26 @@ impl ContainerChainSpawner {
                 )
                 .await?;
 
+            // Generate genesis hash to compare against container client's genesis hash
+            let container_preloaded_genesis = container_chain_cli.preloaded_chain_spec.unwrap();
+            let state_version =
+                Cli::native_runtime_version(&container_preloaded_genesis.cloned_box())
+                    .state_version();
+            let block: Block = generate_genesis_block(&*container_preloaded_genesis, state_version)
+                .map_err(|e| format!("{:?}", e))?;
+            let chain_spec_genesis_hash = block.header().hash();
+
+            let container_client_genesis_hash = container_chain_client.chain_info().genesis_hash;
+
+            if container_client_genesis_hash != chain_spec_genesis_hash {
+                log::info!(
+                    "Container genesis {:?} different from chain spec genesis {:?} - Deleting container db", 
+                    container_client_genesis_hash, 
+                    chain_spec_genesis_hash
+                );
+                delete_container_chain_db(&db_path);
```

Comment:
This won't work because the node has already been started, we need to stop the container chain, delete the db, and start it again.

Reply:
I pushed a commit to your branch to fix this, you can now return an error "NeedsRestartAndDbRemoval" and the container chain will delete the database and restart. Feel free to modify it. Also I merged master and there are some compile errors because of the 1.1.0 upgrade, sorry!